scm-core/src/main/java/sonia/scm/user/User.java

```diff
@@ -114,407 +94,123 @@ public User(String name, String displayName, String mail, String password, Strin
-           && Objects.equal(active, other.active)
-           && Objects.equal(password, other.password)
-           && Objects.equal(creationDate, other.creationDate)
-           && Objects.equal(lastModified, other.lastModified)
-           && Objects.equal(properties, other.properties);
+      && Objects.equal(displayName, other.displayName)
+      && Objects.equal(mail, other.mail)
+      && Objects.equal(type, other.type)
+      && Objects.equal(active, other.active)
+      && Objects.equal(password, other.password)
+      && Objects.equal(creationDate, other.creationDate)
+      && Objects.equal(lastModified, other.lastModified)
+      && Objects.equal(properties, other.properties);
   }
 
-  /**
-   * {@inheritDoc}
-   *
-   *
-   * @return
-   */
   @Override
-  public int hashCode()
-  {
+  public int hashCode() {
     return Objects.hashCode(name, displayName, mail, type, password,
-                            active, creationDate, lastModified, properties);
+      active, creationDate, lastModified, properties);
   }
 
-  /**
-   * {@inheritDoc}
-   *
-   *
-   * @return
-   */
   @Override
-  public String toString()
-  {
+  public String toString() {
     String pwd = (password != null)
-                 ? "(is set)"
-                 : "(not set)";
+      ? "(is set)"
+      : "(not set)";
 
     //J-
     return MoreObjects.toStringHelper(this)
-            .add("name", name)
-            .add("displayName",displayName)
-            .add("mail", mail)
-            .add("password", pwd)
-            .add("type", type)
-            .add("active", active)
-            .add("creationDate", creationDate)
-            .add("lastModified", lastModified)
-            .add("properties", properties)
-            .toString();
+      .add("name", name)
+      .add("displayName", displayName)
+      .add("mail", mail)
+      .add("password", pwd)
+      .add("type", type)
+      .add("active", active)
+      .add("creationDate", creationDate)
+      .add("lastModified", lastModified)
+      .add("properties", properties)
+      .toString();
     //J+
   }
 
-  //~--- get methods ----------------------------------------------------------
-
-  /**
-   * Method description
-   *
-   *
-   * @return
-   */
-  public Long getCreationDate()
-  {
-    return creationDate;
-  }
-
-  /**
-   * Method description
-   *
-   *
-   * @return
-   */
-  public String getDisplayName()
-  {
-    return displayName;
-  }
-
-  /**
-   * Method description
-   *
-   *
-   * @return
-   */
-  @Override
-  public String getId()
-  {
-    return name;
-  }
-
-  /**
-   * Method description
-   *
-   *
-   * @return
-   */
   @Override
-  public Long getLastModified()
-  {
-    return lastModified;
-  }
-
-  /**
-   * Method description
-   *
-   *
-   * @return
-   */
-  public String getMail()
-  {
-    return mail;
+  public boolean isValid() {
+    return ValidationUtil.isNameValid(name) && Util.isNotEmpty(displayName)
+      && Util.isNotEmpty(type)
```

Comment:
Is that still the case? Do i have to set a deprecated field? Will something break if I don't do it?